app/oauth.go

```diff
@@ -354,8 +354,13 @@ func (a *App) newSession(appName string, user *model.User) (*model.Session, *mod
 
 func (a *App) newSessionUpdateToken(appName string, accessData *model.AccessData, user *model.User) (*model.AccessResponse, *model.AppError) {
 	// Remove the previous session
-	<-a.Srv.Store.Session().Remove(accessData.Token)
-
+	schan := make(chan *model.AppError, 1)
+	go func() {
+		removeErr := a.Srv.Store.Session().Remove(accessData.Token)
+		schan <- removeErr
+		close(schan)
+	}()
+	<-schan
```

Comment:
I have a question.. What is being done with the data being read from the channel here??